torchdata/nodes/map.py

```diff
@@ -192,6 +192,12 @@ def __init__(
         if self.in_order:
             self._sort_thread.start()
 
+        # Try and get initial snapshot
+        while self._snapshot is None:
+            self._maybe_update_snapshot(-1)
```

Comment:
Is it possible in any scenario _maybe_update_snapshot fails to update self._snapshot and we end up looping?

Reply:
Yes good call out, eg if the worker function hits a segfault. we can add a check here